From b1552d248d33b4f956a2034e5275f5e75dc09256 Mon Sep 17 00:00:00 2001
From: "ocean.wy" <ocean.wy@alibaba-inc.com>
Date: Tue, 15 Jul 2025 22:23:51 +0800
Subject: [PATCH 2/2] ban creating log table

---
 .../fluss/server/coordinator/CoordinatorService.java        | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java b/fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java
index b1d588fe..76f01306 100644
--- a/fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java
+++ b/fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java
@@ -247,6 +247,12 @@ public final class CoordinatorService extends RpcServiceBase implements Coordina
         // apply system defaults if the config is not set
         tableDescriptor = applySystemDefaults(tableDescriptor);
 
+        if (!tableDescriptor.hasPrimaryKey()) {
+            throw new InvalidTableException(
+                    "Currently, Only primary-key table is support in this cluster. if you want "
+                            + "use log table, please contact 温粥(263086) for help.");
+        }
+
         // the distribution and bucket count must be set now
         //noinspection OptionalGetWithoutIsPresent
         int bucketCount = tableDescriptor.getTableDistribution().get().getBucketCount().get();
-- 
2.33.0

